manager/state/raft/raft.go

```diff
@@ -1206,10 +1206,63 @@ func (n *Node) applyRemoveNode(cc raftpb.ConfChange) (err error) {
 	// If the node from where the remove is issued is
 	// a follower and the leader steps down, Campaign
 	// to be the leader.
-	if cc.NodeID == n.Leader() {
+
+	members := n.cluster.Members()
+	if cc.NodeID == n.Leader() && !n.IsLeader() {
 		if err = n.Campaign(n.Ctx); err != nil {
 			return err
 		}
+	} else if cc.NodeID == n.Config.ID && n.IsLeader() && len(members) == 2 {
+		// if there are only 2 nodes in the cluster, and leader is leaving
+		// before closing the connection, leader has to ensure that follower gets
+		// noticed about this raft conf change commit. Otherwise, follower would
+		// assume there are still 2 nodes in the cluster and won't get elected
+		// into the leader by acquiring the majority (2 nodes)
+		//
+		// adopt the best-effort strategy to send this notice, because the raft conf change
+		// already gets committed locally, there is no approach to revert the change
+
+		status := n.Node.Status()
+		var conn *membership.Member
+
+		// get peer
+		for id, member := range members {
+			if id == n.Config.ID {
+				continue
+			}
+
+			conn = member
+			break
+		}
+
+		// construct msg to inform peer of the commit of raft conf change
+		m := raftpb.Message{Type: raftpb.MsgApp,
+			To: conn.RaftID, From: n.Config.ID, Term: status.Term, LogTerm: status.Term,
+			Index: status.Commit, Commit: status.Commit}
+
+		// send with best-effort strategy
+		ctx := context.Background()
+	Loop:
+		for {
```

Comment:
I don't think it makes sense to have a loop here. grpc will automatically retry if it can't reach the remote host (it has a weird asynchronous `Dial` function). If we want to allow a few retries, we should just use a longer duration than `sendTimeout` - perhaps `sendTimeout * 10` or something like that. But I think we absolutely want to put a bound on the number of retries.